builder/azure/common/template/template.go

```diff
@@ -90,11 +90,20 @@ type Properties struct {
 	Sku                          *Sku                                `json:"sku,omitempty"`
 	UserData                     *string                             `json:"userData,omitempty"`
 	//StorageProfile3              *compute.StorageProfile             `json:"storageProfile,omitempty"`
-	StorageProfile *StorageProfileUnion    `json:"storageProfile,omitempty"`
-	Subnets        *[]network.Subnet       `json:"subnets,omitempty"`
-	SecurityRules  *[]network.SecurityRule `json:"securityRules,omitempty"`
-	TenantId       *string                 `json:"tenantId,omitempty"`
-	Value          *string                 `json:"value,omitempty"`
+	StorageProfile          *StorageProfileUnion    `json:"storageProfile,omitempty"`
+	Subnets                 *[]network.Subnet       `json:"subnets,omitempty"`
+	SecurityRules           *[]network.SecurityRule `json:"securityRules,omitempty"`
+	TenantId                *string                 `json:"tenantId,omitempty"`
+	Value                   *string                 `json:"value,omitempty"`
+	Publisher               *string                 `json:"publisher,omitempty"`
```

Comment:
maybe add a comment line //Custom Script extension related properties
as publisher is quite an overaloaded term :(